src/Task1/WordFrequencyCounter.java

```diff
@@ -0,0 +1,117 @@
+package Task1;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class WordFrequencyCounter {
+    public static void main(String[] args) {
+        // Part 1
+        // Find and save song as a String
+
+        String song = "Oh, I get by with a little help from my friends\n" +
+                "Mm, I get high with a little help from my friends\n" +
+                "Mm, I'm gonna try with a little help from my friends";
+
+        // Remove commas, newline characters, and convert to lowercase
+        song = song.replaceAll("[,\\n]", "").toLowerCase();
+
+        // Split the song into an array of words
+        String[] wordsArray = song.split(" ");
+
+        // Create an array to store word frequencies
+        String[] uniqueWordsArray = new String[wordsArray.length];
+        int[] wordFrequencies = new int[wordsArray.length];
+
+        // Count word frequencies and print the results
+        int uniqueWordCount = 0;
+        for (String word : wordsArray) {
+            int index = indexOfWord(uniqueWordsArray, word, uniqueWordCount);
+            if (index == -1) {
+                uniqueWordsArray[uniqueWordCount] = word;
+                wordFrequencies[uniqueWordCount] = 1;
+                uniqueWordCount++;
+            } else {
+                wordFrequencies[index]++;
+            }
+        }
+
+        // Print unique words and their frequencies
+        for (int i = 0; i < uniqueWordCount; i++) {
+            System.out.println(uniqueWordsArray[i] + ": " + wordFrequencies[i]);
+        }
+
+        // Part 2
+        List<String> wordsList = new ArrayList<>();
+        Set<String> uniqueWordsSet = new HashSet<>();
+
+        // Update Part 1 to use a List to count words
+        for (String word : wordsArray) {
+            if (!uniqueWordsSet.contains(word)) {
+                wordsList.add(word);
+                uniqueWordsSet.add(word);
+            }
+        }
+
+        // Print unique words without duplicates
+        System.out.println("Unique Words: " + wordsList);
```

Comment:
You probably meant to use the uniqueWordsSet, since wordsList is an array so it'll save all the words with duplicates.
```suggestion
        // Print unique words without duplicates
        System.out.println("Unique Words: " + uniqueWordsSet);
```